lib/saml_idp_constants.rb

```diff
@@ -47,10 +52,12 @@ module Constants
         LOA3_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL2,
         IAL1_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL1,
         IAL2_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL2,
+        IAL2_BIO_PREFERRED_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL2,
+        IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL2,
         IALMAX_AUTHN_CONTEXT_CLASSREF => ::Idp::Constants::IAL_MAX,
       }.freeze
 
-      AUTHN_CONTEXT_IAL_TO_CLASSREF = {
+      AUTHN_CONTEXT_IAL_TO_DEFAULT_CLASSREF = {
```

Comment:
The references to this primarily use it to set a default ACR value based on `ServiceProvider.ial` (or `ServiceProviderIdentity.ial`) which is stored as an integer.

Here are the usages:
```ruby
# app/controllers/concerns/saml_idp_auth_concern.rb
    SamlIdpAuthConcern.default_ial_context()
        118 Saml::Idp::Constants::AUTHN_CONTEXT_IAL_TO_DEFAULT_CLASSREF[saml_request_service_provider.ial]
# app/models/federated_protocols/saml.rb
    Saml.default_authn_context()
        56  ::Saml::Idp::Constants::AUTHN_CONTEXT_IAL_TO_DEFAULT_CLASSREF[current_service_provider.ial]
# app/services/attribute_asserter.rb
    AttributeAsserter.sp_ial()
        167 Saml::Idp::Constants::AUTHN_CONTEXT_IAL_TO_DEFAULT_CLASSREF[service_provider.ial]
```

The only usage that is a bit tricky is in `AttributeAsserter`. I guess my question would be: since we'll have (temporarily) multiple variants of an IAL2 ACR value, using the stored IAL "level" to get anything but a default ACR value for that level doesn't make sense, right? 

I'll revert the changes as this mostly affects code related to SAML anyway and would need to be paired with changes in `saml_idp`.

Reply:
Reverted.